frontend/src/metabase/nav/components/AppBar/AppBar.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { ReactNode, useCallback, useMemo, useState } from "react";
+import { t } from "ttag";
+import { isSmallScreen } from "metabase/lib/dom";
+import { isMac } from "metabase/lib/browser";
+import LogoIcon from "metabase/components/LogoIcon";
+import Tooltip from "metabase/components/Tooltip";
+import { CollectionId } from "metabase-types/api";
+import NewItemButton from "../NewItemButton";
+import SearchBar from "../SearchBar";
+import SidebarButton from "../SidebarButton";
+import CollectionBreadcrumbs from "../../containers/CollectionBreadcrumbs";
+import {
+  AppBarRoot,
+  LeftContainer,
+  MiddleContainer,
+  RightContainer,
+  SearchBarContainer,
+  SearchBarContent,
+  SidebarButtonContainer,
+  LogoLink,
+  LogoLinkContainer,
+  CollectionBreadcrumbsContainer,
+} from "./AppBar.styled";
+
+interface AppBarProps {
+  collectionId?: CollectionId;
+  isNavBarOpen?: boolean;
+  isNavBarVisible?: boolean;
+  isSearchVisible?: boolean;
+  isNewButtonVisible?: boolean;
+  isCollectionPathVisible?: boolean;
+  onToggleNavbar: () => void;
+  onCloseNavbar: () => void;
+}
+
+const AppBar = ({
```

Comment:
The code is almost the same. I've only renamed a few things and reworked `HomepageLink`.